api/v3/utils.php

```diff
@@ -461,6 +461,382 @@ function _civicrm_api3_store_values(&$fields, &$params, &$values) {
 /**
  * Get function for query object api.
  *
+ * This is a simple get function, but it should be usable for any kind of
+ * entity. I created it to work around CRM-16036.
+ *
+ * @param string $dao_name
+ *   Name of DAO
+ * @param array $params
+ *   As passed into api get function.
+ * @param bool $isFillUniqueFields
+ *   Do we need to ensure unique fields continue to be populated for this api? (backward compatibility).
+ * @param array $extraMysql
```

Comment:
For the scheduled reminder refactor, we also had the need for different parts of the code to mix in their own bits of SQL. The updated `Select` class (now merged into 4.6) can do things like this:

```php
  $fragment = CRM_Utils_SQL_Select::fragment();
    ->join(...)
    ->where(...)
    ->groupBy(...);
  $realQuery = CRM_Utils_SQL_Select::from(...);
    ->merge($fragment);
```

This is better than passing around an `$extraMysql` array for a few reasons:
  * IDE support (autocomplete/drilldown/docs) works with $fragment.
  * You can use expressions like `#actContactType`, `@actContactType`, or `!actContactType` to handle interpolation / escaping / imploding (which makes it harder to accidentally let in a SQL injection like above).